forte/processors/data_augment/algorithms/single_annotation_op.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2022 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+SingleAnnotationAugmentOp is the extension of the BaseDataAugmentationOp.
+The aim of this Op is to increase ease of use and reduce freedom of
+operations that can be done on a data pack. This Op only allows the
+implementation of those augmentation algorithms that process one
+annotation at a time. This annotation can be one Token or even one
+Sentence. But once declares, the augmentation will happen only
+on the specified annotations.
+"""
+from typing import (
+    Tuple,
+    Dict,
+    Any,
+)
+from abc import abstractmethod
+from forte.data.data_pack import DataPack
+from forte.data.ontology.top import Annotation
+from forte.utils.utils import get_class
+from forte.processors.data_augment.algorithms.base_data_augmentation_op import (
+    BaseDataAugmentationOp,
+)
+
+
+__all__ = ["SingleAnnotationAugmentOp"]
+
+
+class SingleAnnotationAugmentOp(BaseDataAugmentationOp):
+    r"""
+    This class extends the `BaseDataAugmentationOp` to
+    only allow augmentation of one annotation at a time.
+    We leave the :func:`single_token_augment` method to
+    be implemented by subclasses.
+    """
+
+    def augment(self, data_pack: DataPack) -> bool:
+        r"""
+        This method is implemented to restrict the flexibility
```

Comment:
Similar to above, this docstring is meant to be read by developers: `implemented to restrict the flexibility` is not something they care.

Reply:
If the purpose of this function is not to be implemented, it should also be described clearly.